### PR TITLE
run listeners from RF report file

### PR DESCRIPTION
**Pull request by egibbm**
We have built a tool by extending your agent to allow posting results to reportportal using RF output.xml.

From the command's doc:
```
This tool replays a Robot Framework test session using the output XML file
with ReportPortal's listener agent injected without actually doing the test.
This will allow all the test results to be sent to ReportPortal, after the
test is run. By doing this, we will not disturb the timing when the test is
run. It also has the benefit of using the test result generated after running
multiple parallel robot testing (eg. by using pabot).
```

This will help solve similar issue as  #65.

Let me know if you're interested in merging this to yours. I'll be glad to help.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 from setuptools import setup, find_packages
 
 
-__version__ = '5.0.3'
+__version__ = '5.0.3-postreport'
```
Comment (iivanou) [inline setup.py:4]: Change version to 5.0.4

File: robotframework_reportportal/variables.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 
+_variables = {}
+
 
 def get_variable(name, default=None):
-    return BuiltIn().get_variable_value("${" + name + "}", default=default)
+    try:
+        return BuiltIn().get_variable_value("${" + name + "}", default=default)
+    except RobotNotRunningError:
+        try:
```
Comment (iivanou) [inline robotframework_reportportal/variables.py:12]: return _variables.get(name, default) without try/except block.

**Review by iivanou — CHANGES_REQUESTED** [review]
Apart from the other comments. What are the major benefits of this implementation comparing to the built-in import functionality of the ReportPortal UI?

image

**Comment by egibbm** [comment]
> Apart from the other comments. What are the major benefits of this implementation comparing to the built-in import functionality of the ReportPortal UI?
> 
> image

the import interface only capable of handling xunit format, which is inadequate / do not provide as complete information as the one provided by listeners. It does not record keywords, so we miss the ability to trace issues/flakiness from the reportportal UI.

**Comment by egibbm** [comment]
btw do we still need to support python2? The check is false positive.
image

**Comment by iivanou** [comment]
> btw do we still need to support python2? The check is false positive.
> image

Unfortunately, there are loads of agent consumers who are still running Python 2. If you wanna add type hints into the project, you can create *.pyi files.

File: README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ First you need to install RobotFramework:
 
     pip install robotframework
 
-The latest stable version of library is available on PyPI:
+The latest stable version of library can be installed directly from github:
```
Comment (iivanou) [inline README.md:19]: The previous message was correct.

**Review by iivanou — APPROVED** [review]

**Pull request merged**